extensions/wikia/ARecoveryEngine/js/SourcePoint/sourcePoint.js

```diff
@@ -0,0 +1,52 @@
+/*global define*/
+define('ext.wikia.aRecoveryEngine.sourcePoint.sourcePoint', [
+	'ext.wikia.adEngine.adContext',
+	'wikia.log',
+	'wikia.window'
+], function (adContext, log, win) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.aRecoveryEngine.sourcePoint.sourcePoint',
+		context = adContext.getContext(),
+		recoverableSlots = [
+			'TOP_LEADERBOARD',
+			'TOP_RIGHT_BOXAD',
+			'LEFT_SKYSCRAPER_2',
+			'LEFT_SKYSCRAPER_3',
+			'INCONTENT_BOXAD_1',
+			'BOTTOM_LEADERBOARD',
+			'GPT_FLUSH'
+		];
+
+	/**
+	 * SourcePoint can be enabled only if PF recovery is disabled
+	 *
+	 * @returns {boolean}
+	 */
+	function isEnabled() {
+		var enabled = !!context.opts.sourcePointRecovery && !context.opts.pageFairRecovery;
```

Comment:
Let's make sure we document this logic outside the code: it's possible to overwrite SourcePoint recovery by enabling PageFair recovery :smiley_cat: